app/models/person.rb

```diff
@@ -112,8 +112,19 @@ def to_param
     username
   end
 
-  def self.find(username)
-    super(self.find_by_username(username).try(:id) || username)
+  def self.find(username_or_id)
+    by_username = self.find_by_username(username_or_id)
```

Comment:
Rails magic 😅

Dynamic finders (`.find_by_xxx`) work for each attribute. And you can even use `and` in the method name, i.e. `.find_by_id_and_username(id, username)`. Cool, isn't it? 😜

But seriously, good catch, I'll change it to `find_by`. In Rails 5 dynamic finders will be removed and instead of `find_by_username(username)` we should use `find_by(username: username)`